sling/myelin/simd-assembler.cc

```diff
@@ -1056,15 +1059,119 @@ class SSEScalarDoubleGenerator : public SIMDGenerator {
   }
 };
 
+// Scalar integer SIMD generator using regular registers.
+class ScalarIntSIMDGenerator : public SIMDGenerator {
+ public:
+  ScalarIntSIMDGenerator(MacroAssembler *masm, Type type)
+      : SIMDGenerator(masm, false), type_(type) {}
+
+  // Uses regular registers.
+  int VectorBytes() override { return TypeTraits::of(type_).size(); }
+  int VectorSize() override { return 1; }
+  int Alloc() override { return masm_->rr().alloc().code(); }
+  bool SupportsUnroll() override { return false; }
+
+  void Load(int dst, const Operand &src) override {
+    switch (type_) {
+      case DT_INT8: masm_->movsxbq(reg(dst), src); break;
+      case DT_INT16: masm_->movsxwq(reg(dst), src); break;
+      case DT_INT32: masm_->movsxlq(reg(dst), src); break;
+      case DT_INT64: masm_->movq(reg(dst), src); break;
+      default: LOG(FATAL) << "Unsupported integer type: " << type_;
+    }
+  }
+
+  void Store(const Operand &dst, int src) override {
+    switch (type_) {
+      case DT_INT8: masm_->movb(dst, reg(src)); break;
+      case DT_INT16: masm_->movw(dst, reg(src)); break;
+      case DT_INT32: masm_->movl(dst, reg(src)); break;
+      case DT_INT64: masm_->movq(dst, reg(src)); break;
+      default: LOG(FATAL) << "Unsupported integer type: " << type_;
+    }
+  }
+
+  void Zero(int r) override {
+    masm_->xorq(reg(r), reg(r));
+  }
+
+  void Add(int dst, int src1, int src2) override {
+    if (dst == src1) {
+      masm_->addq(reg(dst), reg(src2));
+    } else if (dst == src2) {
+      masm_->addq(reg(dst), reg(src1));
+    } else {
+      masm_->movq(reg(dst), reg(src1));
+      masm_->addq(reg(dst), reg(src2));
+    }
+  }
+
+  void Add(int dst, int src1, const jit::Operand &src2) override {
+    if (dst == src1) {
```

Comment:
Seems like testing for the conjunction of dst == src1 and type_ == DT_INT64 in one go would make the code simpler because it would allow merging the two else branches

Reply:
Yes. That's nicer. Done.